src/ml/neural_net/TCMLComputeObjectDetectorDescriptor.m

```diff
@@ -0,0 +1,144 @@
+/* Copyright © 2020 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+#ifdef HAS_MACOS_10_16
```

Comment:
It doesn't build without these guards. If you look at 'ml/neural_network/style_transfer', those files also have guards inspite of building a certain files.

Reply:
then it sounds like something is wrong with the build system. we shouldn't be compiling something specific to an MLCompute implementation on a system that doesn't have MLCompute. we should get to the root of this rather than throwing up guards all over the place.